core/ir/disassemble_shared.c

```diff
@@ -1094,6 +1094,18 @@ internal_instr_disassemble(char *buf, size_t bufsz, size_t *sofar INOUT,
 
     if (!instr_valid(instr)) {
         print_to_buffer(buf, bufsz, sofar, "<INVALID>");
+        if (instr_raw_bits_valid(instr)) {
```

Comment:
I'm confused as to how we'd get into this situation: normally for an unknown encoding we would not have valid raw bits because we would not know the instruction length (except for AArch64).  What use case would end up here?

Reply:
For debugging, the decoder logs some number of raw bytes when it finds an invalid instruction, but it doesn't know the length and just guesses.  Is that what you're trying to do?  I just don't see how the raw bits are valid for that case: I think the decoder does not set the raw bits on a failed decode.

Reply:
For the PT case we would know the exact raw bits, but I see now that that may not be the case in general. Maybe it's better to print the raw bits on failure in the pt decoder itself.

Reply:
> Are you hitting valid opcodes that DR's decoder is not aware of? 

Yes we're hitting stac and clac from #2103.

>  If we were to put it here also,

We can skip putting it here. We can move it to drpt2ir instead where we do know the instr length.

>  Are you using drdecode and for that maybe the existing logging is stubbed out

Yes we're using drdecode in the drpt2ir library. It'll be better to have a visible error on stderr for decoding failures.